### DISCUSSION TITLE
Issue - local images are not displayed

### DISCUSSION THREAD

**Discussion by AM-XR** [ebullient/obsidian-slides-extended#14] [Q&A] [answered]
Before creating a issue I wanted to ask, if its maybe a user error instead a bug.
Loving the plugin so far and would like to do all my presentations with it from now on if possible 🙂   

I am new to the plugin and tried out advanced slides shortly before. 

The issue is that local image embeds dont work at all for me. Not in the preview and also not in the browser and returns the following: 
```
Failed to load resource: the server responded with a status of 404 (Not Found)
```

different formattings i tried that worked with advanced slides: 
```
%% Default %%
![[Out_3.jpg|300x250]]
 
%% Relative Path %%
![](../_ImagesCollection/Out_3.jpg) 
```
![Screenshot 2024-06-02 124259](https://github.com/ebullient/obsidian-slides-extended/assets/80644364/2cd7e040-1d06-41dc-8973-91c0ef0a16f4)

weblinks do work: 
```
![](https://picsum.photos/id/1005/250/250)
```

tried multiple things already:
- reinstalling
- stopping and reenabling the extended slides server
- back and forth testing with advanced slides

best regards,
AM

**Comment by AM-XR** [comment]
After further research, I was able to find out that, unlike Advanced Slides and RevealJS (native), the local server of Extended Slides does not include the images. Via Chrome Dev Tools Sources these are not listed in contrast to the other two. (Should have the same structure then Advanced Slides Example)
- The sample images are located in the Vault folder

![Advanced slides sources](https://github.com/ebullient/obsidian-slides-extended/assets/80644364/49f37337-1f9e-4352-81b6-2a24c8c62472)
![revealJS sources](https://github.com/ebullient/obsidian-slides-extended/assets/80644364/449757eb-75c2-4a72-a911-2bc0995fd674)
![Slides extended sources](https://github.com/ebullient/obsidian-slides-extended/assets/80644364/d8e5930a-e4ff-48a4-b7d6-256b1a4fcc1e)

**Comment by ebullient** [comment]
I am a little confused by this, as I use this extensively _with local images_. So it does work, in at least my case. 

"(Should have the same structure then Advanced Slides Example)" -- that won't be true. This is a newer version of reveal.js, and I've made some changes to how things are done. There is an increasing amount that is not identical to advanced slides.

What would be most helpful here is a zipped up sample vault, with your preferred link settings and a sample presentation. I can use that to create a test to both recreate it, fix it, and ensure it remains fixed. ;)

**Comment by AM-XR** [comment]
Here are two simple vaults. Version two with Advanced Slides is for reference only, as it works for me.
- None of the images are displayed for me
- tested with Firefox and Chrome
- OS: Windows 10
- Is reproducible on a second computer (also Win10)

Hope I can be of help with this. If I can contribute anything else, please let me know :)
[SampleVault_AdvancedSlides_Reference.zip](https://github.com/user-attachments/files/15535579/SampleVault_AdvancedSlides_Reference.zip)
[SampleVault_SlidesExtended.zip](https://github.com/user-attachments/files/15535580/SampleVault_SlidesExtended.zip)

**Comment by AM-XR** [comment]
Hello ebullient,

I wanted to give the new version another go and tested the issue of no local images showing up again. The issue seems to persist. 

Do you have an idea how I could support you with additional data or tests?

Did the testvaults maybe help to reproduce the issue? For me its reproducible on a second win10 pc that has no custom configurations, just to exclude that possibility. 

best regards,
AM

**Reply by ebullient** [reply]
Your test vaults both work for me without an issue, either within the preview, or when launched in the external browser.

I'll note that the path you're using for local images is a little confusing (your file is in the root of the vault, and you're using a `../` prefix, which is an incorrect relative path.
It should be this, based on what you've provided:

```
### relative path
img_a no sizing
![](./images/img_a.jpg) 

img_b with sizing
![|300x300](./images/img_b.jpg) 
```

For your sample vault, I see this when I inspect the rendered page (which is correct):
<img width="223" alt="image" src="https://github.com/user-attachments/assets/a88a0a6b-25e0-44ce-9957-d258b0a3d4d8">

**Reply by ebullient** [reply]
Based on #23, windows may be requiring you to use localhost in the URL instead. This is hard for me to verify, as I don't run windows.

**Reply by AM-XR** [reply]
Thank you for the quick response.

I have now also tried the test with localhost and the changed relative path. Unfortunately, neither brings about any change.

Here is a screenshot from the Developer Tools Sources. The images don't seem to be on my server at all.

I can reproduce it on my second computer (Win10), but unfortunately I do not have a Mac or Linux operating system available to test whether it is a Windows-only problem.

Best regards,
AM

![image](https://github.com/user-attachments/assets/a8f48a98-4f9b-4ecc-9af6-e6c56d7faeab)
![plXUzn4lN4](https://github.com/user-attachments/assets/ea85ec4d-db80-43f5-bbc8-1740547cd615)

**Comment by ebullient** [comment, marked as answer]
Closing this -> issue #24
I've attempted a fix, if you could try again when you have a chance, open a new issue with whatever is still happening. ;)
